packages/core/src/util/ids.ts

```diff
@@ -0,0 +1,29 @@
+import { UISchemaElement } from '..';
+
+const idMappings: Map<UISchemaElement, string> = new Map<UISchemaElement, string>();
+
+export const createId = (element: UISchemaElement, proposedID: string) => {
+  let tries = 0;
+  while (!isUniqueId(proposedID, tries)) {
+    tries++;
+  }
+  const newID = makeId(proposedID, tries);
+  idMappings.set(element, newID);
+  return newID;
+};
+
+export const removeId = uischema => idMappings.delete(uischema);
```

Comment:
Add type annotation to `uischema`. Can we add a test case for this, i.e. upon unmounting the id should be deleted?